### PR TITLE
Fix components extending beyond screen dimensions on mobile view

### LINKED ISSUE #2241: Mobile: Long dialog titles cause the generic dialog components to extend beyond the screen dimensions on mobile view
The generic `ConfirmDialog` and `PromptDialog` components both suffer from this issue, which affects all existing dialogs where the title is fairly long.

image

### PR DESCRIPTION
**Pull request by Johnwz123**
### Description

Closes #2241 

The following components extend beyond the screen dimensions on mobile view:
- `ConfirmDialog` and `PromptDialog`
- SICP "Text Search" and "Index Search" search bar
- Assessment `OverviewCard`

### Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] Code quality improvements

### How to test

On a Mobile view:

`ConfirmDialog` and `PromptDialog`
1. In Playground, switch to Source 2, 3 or 4 to access and "Enable Folder mode"
2. Create a "New File" or "New Directory" with a long name.
3. Try to "Delete" the file and the Dialog box will appear.

SICP "Text Search" and "Index Search" search bar
1. In SICP JS, click "Text Search" or "Index Search" to display the search bar.

Assessment `OverviewCard`
1. In the Assessments tabs (e.g. Missions, Quests), the Assessment OverviewCard layout should display with the icon on top of the description.

### Checklist

- [x] I have tested this code
- [ ] I have updated the documentation

### CODE REVIEW

**Review by sayomaki — COMMENTED** [review]
Thanks for the changes! I have also added some minor style updates, and have fixed the formatting as well as tests for you.

File: src/styles/_sicp.scss [hunk 1]
```diff
@@ -353,3 +353,10 @@ $sicp-content-lr-padding: 6em;
     width: 70% !important;
   }
 }
+
+.bp4-omnibar {
+  @media only screen and (max-width: 768px) {
+    left: 1%;
+    width: 98%;
+  }
+}
```
Comment (RichDom2185) [inline src/styles/_sicp.scss:362]: Please do not use Blueprint's CSS API (see #2722, https://github.com/source-academy/frontend/pull/2775#discussion_r1487232448).

Also, didn't we upgrade to Blueprint v5? Does this style make a difference?

File: src/styles/ConfirmDialog.module.scss [hunk 2]
```diff
@@ -2,4 +2,18 @@
...
+    width: 98%;
+  }
+}
+
+.PromptDialog {
+  @media only screen and (max-width: 500px) {
+    // Set width to 98% when screen width is less than 500px
+    // 500px is the default width of the component
+    width: 98%;
+  }
```
Comment (RichDom2185) [inline src/styles/ConfirmDialog.module.scss:18]: I don't see this class being used anywhere?

**Review by RichDom2185 — CHANGES_REQUESTED** [review]

File: src/styles/ConfirmDialog.module.scss (hunk 2 above)
Comment (Johnwz123) [reply src/styles/ConfirmDialog.module.scss]: Yes, my bad. The PromptDialog component uses the ConfirmDialog component as well.

File: src/styles/_sicp.scss (hunk 1 above)
Comment (Johnwz123) [reply src/styles/_sicp.scss]: I have added a class to the search bar component and updated the style to reference the class.

**Review by RichDom2185 — COMMENTED** [review]
Please update the test snapshots, thanks!

**Review by sayomaki — APPROVED** [review]
Looks good to me, thanks!

**Review by RichDom2185 — APPROVED** [review]
LGTM, thanks!

**Pull request merged**